src/main/java/com/hcl/appscan/jenkins/plugin/builders/AppScanBuildStep.java

```diff
@@ -294,8 +295,23 @@ private void perform(Run<?,?> build, Launcher launcher, TaskListener listener) t
     	m_authProvider = new JenkinsAuthenticationProvider(m_credentials, build.getParent().getParent());
     	final IProgress progress = new ScanProgress(listener);
     	final boolean suspend = m_wait;
-    	final IScan scan = ScanFactory.createScan(getScanProperties(build, listener), progress, m_authProvider);
-        
+        Map<String, String> properties = getScanProperties(build,listener);
+    	final IScan scan = ScanFactory.createScan(properties, progress, m_authProvider);
+        JenkinsAuthenticationProvider checkAppScan360 = (JenkinsAuthenticationProvider) m_authProvider;
+        if(checkAppScan360.isAppScan360()) {
+            if (m_type.equals("Dynamic Analyzer")) {
+                throw new AbortException(Messages.error_dynamic_analyzer_AppScan360());
+            } else if (m_intervention) {
+                progress.setStatus(new Message(Message.WARNING, Messages.warning_allow_intervention_AppScan360()));
+            } else if (properties.get("openSourceOnly") != null) {
```

Comment:
I suggest having this as the 2nd check or remove the "else", so it's all "if" conditions. As it is, we should be throwing an exception if the user selected open source only, but if they also chose to allow intervention, we would continue with the open source only scan (i.e. we wouldn't throw).

Reply:
removed the "else" from the conditions.